k_combinat_for_sage/all.py

```diff
@@ -170,30 +170,30 @@ def number_of_noninversions(lis):
                     num += 1
         return num
     gamma = Composition(gamma)
-    if s.__class__.__name__ in ('SymmetricFunctionAlgebra_monomial_with_category', 'SymmetricFunctionAlgebra_dual_with_category'):
+    if basis.__class__.__name__ in ('SymmetricFunctionAlgebra_monomial_with_category', 'SymmetricFunctionAlgebra_dual_with_category'):
         raise NotImplemented(
             'Straightening does not exist (that i know of) for the monomial basis or the forgotten/dual basis.')
-    elif s.__class__.__name__ == 'HallLittlewood_qp_with_category':
-        return compositional_hall_littlewood_Qp(gamma, t=s.t, base_ring=s.base_ring())
```

Comment:
Why did you delete this line? It is very important. Otherwise, there will be hidden t's in the HLQp terms and the specialization does not work correctly.

Reply:
No reason other than I couldn't understand why you needed it, and I needed to troubleshoot.  I think I understand now.  When you create the parent basis `hall_littlewood(t=val)`, it sets the attribute `.t` equal to val, so here you are making sure that value gets passed along when the new straightened version of the hall littlewood is made.  Correct me if I'm wrong.  You can add this back in with your other changes.

Reply:
Yeah, that is correct!